terraform/ec2/win/main.tf

```diff
@@ -34,66 +58,67 @@ resource "aws_instance" "cwagent" {
-Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
-
-Start-Service sshd
-Set-Service -Name sshd -StartupType 'Automatic'
-
-[System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072
-Set-ExecutionPolicy Bypass -Scope Process -Force; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
-
-choco install git --confirm
-choco install go --confirm
-msiexec /i https://awscli.amazonaws.com/AWSCLIV2.msi  /norestart /qb-
-
-[Environment]::SetEnvironmentVariable("PATH", "C:\ProgramData\chocolatey\bin;C:\Program Files\Git\cmd;C:\Program Files\Amazon\AWSCLIV2\;C:\Program Files\Go\bin;C:\Windows\System32;C:\Windows\System32\WindowsPowerShell\v1.0\", [System.EnvironmentVariableTarget]::Machine)
-</powershell>
-EOF
 
   tags = {
-    Name = "cwagent-integ-test-ec2-windows-${element(split("/", var.test_dir), 3)}-$${module.common.testing_id}"
+    Name = "cwagent-integ-test-ec2-windows-${module.common.testing_id}"
   }
 }
 
 resource "null_resource" "integration_test" {
   depends_on = [aws_instance.cwagent]
+
   # Install software
+  connection {
+    type     = "winrm"
+    user     = "Administrator"
+    password = rsadecrypt(aws_instance.cwagent.password_data, local.private_key_content)
+    host     = aws_instance.cwagent.public_dns
+  }
+
+  provisioner "file" {
+    source      = "${var.test_dir}/${local.final_validator_config}"
+    destination = "${local.instance_temp_directory}/${local.final_validator_config}"
+  }
+
+  provisioner "file" {
+    source      = "${var.test_dir}/${local.cloudwatch_agent_config}"
+    destination = "${local.instance_temp_directory}/${local.cloudwatch_agent_config}"
+  }
+
+  # Install agent binaries
   provisioner "remote-exec" {
     inline = [
-      "start /wait timeout 120",                             //Wait some time to ensure all binaries have been downloaded
-      "call %ProgramData%\\chocolatey\\bin\\RefreshEnv.cmd", //Reload the environment variables to pull the latest one instead of restarting cmd
-      "set AWS_REGION=${var.region}",
+      "git clone --branch ${var.github_test_repo_branch} ${var.github_test_repo}",
       "aws s3 cp s3://${var.s3_bucket}/integration-test/packaging/${var.cwa_github_sha}/amazon-cloudwatch-agent.msi .",
       "start /wait msiexec /i amazon-cloudwatch-agent.msi /norestart /qb-",
-      "echo clone and install agent",
-      "git clone ${var.github_test_repo}",
+
+    ]
+  }
+
+  provisioner "remote-exec" {
+    inline = [
       "cd amazon-cloudwatch-agent-test",
-      "git reset --hard ${var.cwa_test_github_sha}",
-      "echo run tests with the tag integration, one at a time, and verbose",
-      "echo run sanity test && go test ./test/sanity -p 1 -v",
-      "go test ${var.test_dir} -p 1 -timeout 30m -v"
+      "go run ./validator/main.go --validator-config=${local.instance_temp_directory}/${local.final_validator_config} --preparation-mode=true",
```

Comment:
Is this running twice?

Reply:
No it is different
```
go run ./validator/main.go --validator-config=${local.instance_temp_directory}/${local.final_validator_config} --preparation-mode=false
```
with the `preparation-mode`  false while the other one is true.